### ISSUE TITLE
Failing on non-existent subdir

### ISSUE THREAD

**Issue by xhochy** [QuantStack/mamba#180]
_This worked before the weekend with the then on-conda-forge-published `mamba` releases`._

`conda` can deal with non-existent subdirs in a channel and simply ignores them. This happens for me with a private channel that only has `noarch` but not `osx-64` as a subdir.

Steps to reproduce:

```
% CONDA_SUBDIR=foo-64 conda create -n testenv python=3.8                                                    :(
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - python=3.8
```

```
% CONDA_SUBDIR=foo-64 mamba create -n testenv python=3.8                                                    :(

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        Supported by @QuantStack

        GitHub:  https://github.com/QuantStack/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████

pkgs/main/noarch         [====================] (00m:00s) No change
pkgs/r/noarch            [====================] (00m:00s) No change
conda-forge/noarch       [====================] (00m:00s) Done
pkgs/main/foo-64         [>                   ] (00m:26s) Failed (404), retry in 18s
conda-forge/foo-64       [>                   ] (00m:26s) Failed (404), retry in 18s
pkgs/r/foo-64            [>                   ] (00m:26s) Failed (404), retry in 18s

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/conda/exceptions.py", line 1079, in __call__
        return func(*args, **kwargs)
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/mamba/mamba.py", line 741, in exception_converter
        raise e
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/mamba/mamba.py", line 735, in exception_converter
        exit_code = _wrapped_main(*args, **kwargs)
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/mamba/mamba.py", line 705, in _wrapped_main
        exit_code = do_call(args, p)
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/mamba/mamba.py", line 624, in do_call
        exit_code = create(args, parser)
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/mamba/mamba.py", line 553, in create
        install(args, parser, 'create')
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/mamba/mamba.py", line 397, in install
        unknown=index_args['unknown'], prefix=prefix)
      File "/Users/uwe/miniconda3/lib/python3.7/site-packages/mamba/utils.py", line 52, in get_index
        is_downloaded = dlist.download(True)
    RuntimeError: Download failed.
```

```
% conda list | grep mamba
mamba                     0.2.4            py37h7b8d24a_0    conda-forge
% conda list | grep libsolv
libsolv                   0.7.13               hd174df1_2    conda-forge
```

**Comment by wolfv** [comment]
thanks for testing & pointing this out.

sorry for the breakage, will be fixed asap!

**Comment by wolfv** [comment]
Ok I checked, and the logic is apparently that it ignores 404's if the channel isn't noarch, and only raises an error if 404ing on noarch (https://github.com/conda/conda/pull/8154/files). 

We can surely do the same :)

**Comment by wolfv** [comment]
hey @xhochy I just pushed a fix (still need to make a release with it).

This still waits until retry / timeout is done. Do you think it would be better to not wait? I thought it might be nice to wait a little if someone is on a spotty internet connection but maybe it's not worth it to wait on a 404.

**Comment by wolfv** [comment]
hey @xhochy 

I just released 0.2.5 on conda-forge -- that should have a better behavior for this case. Thanks for raising the issue :) let me know how testing goes!

**Comment by xhochy** [comment]
Everything fine, I'm back on speedy installation of packages again! Thanks!
